gravitee-apim-cypress/cypress/integration/rest-api/apis/jwt-policy-tests.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  closeApplicationSubscription,
+  createApplication,
+  deleteApplication,
+  subscribeApplication,
+} from '@commands/management/application-management-commands';
+import { ApplicationFakers } from '@fakers/applications';
+import { requestGateway } from 'support/common/http.commands';
+import * as faker from 'faker';
+import { PolicyFakers } from '@fakers/policies';
+
+context('Create and test a JWT policy', () => {
+  let createdApi: ApiImport;
+  let applicationId: string;
+  let planId: string;
+  let subscriptionId: string;
+  let apiKey: string;
+  const clientId = `cypress_test_client${faker.random.alpha({ count: 5 })}`;
+  const secret = faker.random.alpha({ count: 32 });
+  const payload = { client_id: clientId, exp: 1900000000 };
+  const jwtToken = jwt.sign(payload, secret, { noTimestamp: true });
+
+  before(() => {
+    const jwtPolicy = PolicyFakers.jwtPolicy(secret);
+    const fakeFlow = ApiImportFakers.flow({ pre: [jwtPolicy] });
+    const fakePlan = ApiImportFakers.plan({ security: PlanSecurityType.API_KEY, flows: [fakeFlow] });
+    const fakeApi = ApiImportFakers.api({ plans: [fakePlan] });
+    importCreateApi(API_PUBLISHER_USER, fakeApi)
+      .ok()
+      .its('body')
+      .then((api) => {
+        createdApi = api;
+        planId = createdApi.plans[0].id;
+        publishPlan(API_PUBLISHER_USER, createdApi.id, planId).ok();
+      })
+      .then(() => {
+        const fakeApplication = ApplicationFakers.application({ settings: { app: { client_id: clientId } } });
+        createApplication(API_PUBLISHER_USER, fakeApplication).created().its('body');
+      })
+      .then((application) => {
+        applicationId = application.id;
+        subscribeApplication(API_PUBLISHER_USER, applicationId, planId)
+          .created()
+          .then((response) => (subscriptionId = response.body.id));
+      })
+      .then(() => {
+        deployApi(API_PUBLISHER_USER, createdApi.id).ok();
+        startApi(API_PUBLISHER_USER, createdApi.id).noContent();
+        getApiKeys(API_PUBLISHER_USER, createdApi.id, subscriptionId)
+          .ok()
+          .then((response) => (apiKey = response.body[0].key));
+      });
+  });
+
+  it('should fail to call API endpoint without JWT token and API-Key', () => {
+    requestGateway(
+      {
+        method: 'GET',
+        url: `${Cypress.env('gatewayServer')}${createdApi.context_path}`,
+      },
+      {
+        validWhen: (response) => {
+          return response.status === 401;
+        },
+      },
+    )
+      .unauthorized()
+      .should((response: Cypress.Response<any>) => {
+        expect(response.body.message).to.equal('Unauthorized');
+      });
+  });
+
+  it('should fail to call API endpoint with API-Key but without JWT token', () => {
+    requestGateway(
+      {
+        method: 'GET',
+        url: `${Cypress.env('gatewayServer')}${createdApi.context_path}`,
+        headers: {
+          'x-gravitee-api-key': `${apiKey}`,
```

Comment:
For a dedicated "api key plan test",
Maybe we should also test the other way of providing an API key, via the query parameter.

Reply:
Right, I'll include a test for query params.